### PR TITLE
Misc. changes to `nf-core modules install`

### PR DESCRIPTION
**Pull request by ErikDanielsson**
Misc. changes to install:
- Added check that the remote of a module exists.
- Made `--latest` default for `modules install --all`. It is still possble to override with `--sha`
- Remove unnecessary critical error from `nf-core modules install`.
- Added try-except blocks in a few more places.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Documentation in `docs` is updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -991,7 +991,7 @@ There are five flags that you can use with this command:
 * `--latest`: Install the latest version of the module instead of specifying the version using the cli prompt.
 * `--force`: Overwrite a previously installed version of the module.
 * `--sha <commit_sha>`: Install the module at a specific commit from the `nf-core/modules` repository.
-* `--all`: Use this flag to change versions on all installed modules. Has the same effect as running `nf-core modules install --force` on all installed modules.
+* `--all`: Use this flag to change versions on all installed modules. Has the same effect as running `nf-core modules install --force --latest` on all installed modules. To change all modules to a specific version you can run `nf-core modules install --all --sha <commit sha>`.
```
Comment (drpatelh) [inline README.md:994]:
```suggestion
* `--all`: Use this flag to change versions on all installed modules. Has the same effect as running `nf-core modules install --force` on all installed modules. To change all modules to a specific version you can run `nf-core modules install --all --sha <commit sha>`.
```

Comment (drpatelh) [reply README.md:994]: Sorry, I thought we were removing `--latest` altogether as an option?

Comment (ErikDanielsson) [reply README.md:994]: No, the `--latest` flag should be included. If we don't want to remove the cli prompt entirely?

Comment (drpatelh) [reply README.md:994]: Ah, I see. So its one of those options we need to keep for the CLI prompt. Finally got there! 😅 Ok.

**Review by drpatelh — APPROVED** [review]
LGTM!

**Comment by ErikDanielsson** [comment]
Thanks for the review @drpatelh!

**Pull request merged**